lib/messageQueue.js

```diff
@@ -19,36 +21,165 @@ var connected = false;
 module.exports = {
 };
 
-module.exports.init = function(config, processGradingResult, callback) {
-    if (config.sqsConfig) {
-        // set up SQS
+module.exports.init = function(processGradingResult, callback) {
+    if (config.autograderUseAws) {
+        // So, this is terrible, but AWS will look relative to the Node working
+        // directory, not the current directory. So aws-config.json should be
+        // in the project root.
+        if (fs.existsSync('./aws-config.json')) {
+            logger.info('Loading AWS credentials for autograder')
+            AWS.config.loadFromPath('./aws-config.json');
+        } else {
+            logger.info('Missing \'aws-config.json\' in project root; this should only matter for local development')
+        }
         callback(null);
     } else {
         // local dev mode
+        logger.info('Not loading AWS credentials; autograder running locally')
         module.exports.processGradingResult = processGradingResult;
         callback(null);
     }
 };
 
 module.exports.sendToGradingQueue = function(grading_log, submission, variant, question, course, callback) {
-    // Helper function
-    let deleteFolderRecursive = function (path) {
-        if (fs.existsSync(path)) {
-            fs.readdirSync(path).forEach(function (file, index) {
-                let curPath = path + "/" + file;
-                if (fs.lstatSync(curPath).isDirectory()) { // recurse
-                    deleteFolderRecursive(curPath);
-                } else { // delete file
-                    fs.unlinkSync(curPath);
-                }
-            });
-            fs.rmdirSync(path);
-        }
-    };
+    if (config.autograderUseAws) {
+        logger.info('submitting grading job id: ' + grading_log.id);
 
-    if (config.sqsConfig) {
-        // TODO push data to S3 and SQS
         callback(null);
+
+        logger.info(JSON.stringify(submission.submitted_answer))
+
+        if (question.autograding_enabled != true) {
+          // Autograding not specified!
+          logger.info('autograding disabled for job id: ' + grading_log.id);
+
+          // Make the grade a automatic 100
+          let ret = {
+              gradingId: grading_log.id,
+              grading: {
+                  score: 1,
+                  feedback: {msg: "Autograder is not enabled :("},
+              },
+          };
+
+          // Send the grade out for processing and display
+            module.exports.processGradingResult(ret);
+        } else {
+            // We should submit our grading job to AWS
+            // First, construct a local directory to hold the question files
+            const dir = path.join('../autograder_jobs', `job_${grading_log.id}`);
+            fs.mkdirsSync(dir);
```

Comment:
Looks like this code is still undergoing changes so sync calls are fine for now, but when moving to async I suggest using http://caolan.github.io/async/. See https://github.com/PrairieLearn/PrairieLearn/blob/master/server.js#L325 for an example of `async.series`.